tests/publiccloud/ansible_client.pm

```diff
@@ -0,0 +1,47 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Summary: prepare the VM to sync with the ansible_target
+# Maintainer: qa-c@suse.de
+
+use base "consoletest";
+use serial_terminal 'select_serial_terminal';
+use strict;
+use warnings;
+use testapi;
+use mmapi;
+use utils qw(zypper_call);
+use network_utils qw(setup_static_network);
+use lockapi;
+
+sub run {
+    my ($self, $args) = @_;
+    select_serial_terminal;
+
+    record_info('system', script_output('cat /etc/os-release'));
+    setup_static_network(ip => '10.0.2.15/15', gw => '10.0.2.2');
+    record_info('ip', script_output('ip a'));
+    record_info('route', script_output('ip r'));
+    assert_script_run('echo "10.0.2.20  microos" >> /etc/hosts');
+    zypper_call('in -y iputils git');
+
+    assert_script_run('mkdir /root/.ssh');
+    assert_script_run('curl -f -v ' . autoinst_url . '/data/slenkins/ssh/id_rsa > /root/.ssh/id_rsa');
```

Comment:
Please either allow password login and generate the key in the test or at least generate our own key in `data/publiccloud/`

Reply:
Let's please not have pre-generated keys as they always expose a certain security risk. Unless really really really necessary, let's generate per-test key pairs.

Reply:
> Let's please not have pre-generated keys as they always expose a certain security risk. Unless really really really necessary, let's generate per-test key pairs.

IMHO it is too much . we can not apply common security rules into short living test VM running in internal network . Generating per-test key pairs coming with very high price -> in case debug is needed you can access VM only via VNC

Reply:
if we really want to care about security of test VM running on internal network let's hide ssh key in worker config so it will be protected on same level as some other info which has much much more value